foam_controlwrapper/tests/test_foam_interface.py

```diff
@@ -0,0 +1,292 @@
+                patchNameFacesMap[boundary].append(
+                    self._uuidToFoamLabel[face.uid])
+
+            # make compressed list of faces points
+            self.facePoints.append(len(face.points))
+            for puid in face.points:
+                self.facePoints.append(self._uuidToFoamLabel[puid])
+
+        # make points coordinate list
+        self.pointCoordinates = []
+        for point in self.mesh.iter_points():
+            for coord in point.coordinates:
+                self.pointCoordinates.append(coord)
+
+        # make compressed list of cells points
+        self.cellPoints = []
+        for cell in self.mesh.iter_cells():
+            self.cellPoints.append(len(cell.points))
+            for puid in cell.points:
+                self.cellPoints.append(self._uuidToFoamLabel[puid])
+
+        # make patch information
+        self.patchNames = []
+        self.patchFaces = []
+        for patchName in patchNameFacesMap:
+            self.patchNames.append(patchName)
+            self.patchFaces.append(len(patchNameFacesMap[patchName]))
+            for face in patchNameFacesMap[patchName]:
+                self.patchFaces.append(face)
+
+        if not self.patchNames:
+            error_str = 'Could not initialize with mesh  {}. '
+            error_str += 'Mesh has not boundary face definitions.'
+            raise ValueError(error_str.format(self.mesh.name))
+
+        # this to have controlDict file for mesh definition
+        FoamFiles().write_default_files(self.path, 'simpleFoam',
+                                        self.time, False)
+
+        foamface.init(self.name, os.path.abspath(os.path.join(self.path,
+                                                              os.pardir)))
+
+    def tearDown(self):
+        if os.path.exists(self.test_dir):
+            shutil.rmtree(self.test_dir)
+
+    def test_add_mesh(self):
+        """Test addMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+
+    def test_write_mesh(self):
+        """Test writeMesh method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        foamface.writeMesh(self.name)
+
+    def test_get_point_coordinates(self):
+        """Test getPointCoordinates method
+
+        """
+
+        foamface.addMesh(self.name, self.pointCoordinates,
+                         self.cellPoints, self.facePoints,
+                         self.patchNames, self.patchFaces)
+        for point in self.points:
+            coords = foamface.getPointCoordinates(
+                self.name,
+                self._uuidToFoamLabel[point.uid])
+            self.assertEqual(point.coordinates, tuple(coords))
```

Comment:
I think in the current implementation of the openfoam wrapper Points should have:
```python 
self.assertEqual(point.data, DataContainer())
self.assertiInstance(point.data, DataContainer). 
```

similar for the faces and edges, if I am correct.